### PR TITLE
Fix test dependencies are not resolved when running single module

### LINKED ISSUE #19987: Test dependencies are not resolved when running module
**Description:**
When a module uses another module only in test dependencies it is not resolved when building or running the module.

**Steps to reproduce:**
Create a project and create two modules in it say `api` and `testutils`
Make sure the module `api` uses `testutils` only in its test cases.
Try building/running only the api module:

`ballerina run api`

This fails with the following error:
Creating balos
```
	target/balo/api-2019r3-java8-0.1.0.balo
error: /xyz/testutils/0.1.0/testutils.bir not found in /home/riyafa/Documents/Workspaces/xyz/components/backend/target/caches/bir_cache /home/riyafa/.ballerina/bir_cache-1.0.4 /home/riyafa/Programs/Ballerina/bir-cache 
	at ballerina.compiler_backend_jvm:findCacheDirFor(jvm_package_gen.bal:332)
	   ballerina.compiler_backend_jvm:lookupModule(jvm_package_gen.bal:311)
	   ballerina.compiler_backend_jvm:generatePackage(jvm_package_gen.bal:133)
	   ballerina.compiler_backend_jvm:generateDependencyList(jvm_package_gen.bal:120)
	   ballerina.compiler_backend_jvm:generatePackage(jvm_package_gen.bal:142)
	   ballerina.compiler_backend_jvm:generateJarBinary(main.bal:87)
	   ballerina.compiler_backend_jvm:main(main.bal:60)
make: *** [Makefile:27: run-backend] Error 1
```
**Affected Versions:**
1.0.4

### PR DESCRIPTION
**Pull request by Kishanthan**
$subject

Fixes #19987

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Required Balo version update
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [x] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java [hunk 1]
```diff
@@ -74,37 +78,45 @@ public CopyNativeLibTask() {
...
         // Iterate through the imports and copy dependencies.
         HashSet<PackageID> alreadyImportedSet = new HashSet<>();
         for (BLangPackage pkg : moduleBirMap) {
-            // Copy jars from imported modules.
-            copyImportedLibs(pkg.symbol.imports, buildContext.moduleDependencyPathMap.get(pkg.packageID).platformLibs,
-                             buildContext, sourceRootPath, balHomePath, alreadyImportedSet);
+            PackageID packageID = pkg.packageID;
+            BLangPackage bLangPackage = packageCache.get(packageID);
+
+            if (bLangPackage == null || !buildContext.moduleDependencyPathMap.containsKey(packageID)) {
```
Comment (hemikak) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java]: When will bLangPackage be null ?

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunTestsTask.java [hunk 2]
```diff
@@ -49,6 +49,12 @@ public void execute(BuildContext buildContext) {
         // as "." are ignored. This is to be consistent with the "ballerina test" command which only executes tests
         // in packages.
         for (BLangPackage bLangPackage : moduleBirMap) {
+            PackageID packageID = bLangPackage.packageID;
+
+            if (!buildContext.moduleDependencyPathMap.containsKey(packageID)) {
```
Comment (hemikak) [inline cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunTestsTask.java]: Is there are cases when the map doesn't contain packageID

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java (hunk 1 above)
Comment (Kishanthan) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/CopyNativeLibTask.java]: Can be, if it is not there in the package cache.

File: cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunTestsTask.java (hunk 2 above)
Comment (Kishanthan) [reply cli/ballerina-packerina/src/main/java/org/ballerinalang/packerina/task/RunTestsTask.java]: Yes, there was a case when i tested and it threw NPE

**Pull request merged**